### PR TITLE
services: binlog status code, msg, details

### PR DESCRIPTION
**Pull request by zpencer**
These are first class concepts and must exist as top level fields.

### CODE REVIEW

File: services/src/main/java/io/grpc/services/BinlogHelper.java [hunk 1]
```diff
@@ -83,6 +83,13 @@
   private static final Logger logger = Logger.getLogger(BinlogHelper.class.getName());
   private static final boolean SERVER = true;
   private static final boolean CLIENT = false;
+  // This is a special key that is set from above the application API,
+  // despite having a 'grpc-' prefix. Normally 'grpc-' metadata keys are set from within
+  // gRPC.
+  static final Metadata.Key<byte[]> STATUS_DETAILS_KEY =
```
Comment (carl-mastrangelo) [inline services/src/main/java/io/grpc/services/BinlogHelper.java]: Shouldn't this be snuck from `StatusProto`?

Comment (zpencer) [reply services/src/main/java/io/grpc/services/BinlogHelper.java]: The StatusProto version of this turns it into an actual proto object, but we only care about the bytes. Also, that Key is private which is inconvenient to deal with.

Comment (carl-mastrangelo) [reply services/src/main/java/io/grpc/services/BinlogHelper.java]: Can you comment saying the name must be identical to the actual key location?

Comment (zpencer) [reply services/src/main/java/io/grpc/services/BinlogHelper.java]: You mean the fact that "grpc-status-details-bin" is not some arbitrary name, but is instead a special well known key?

Comment (carl-mastrangelo) [reply services/src/main/java/io/grpc/services/BinlogHelper.java]: yes.  It needs to be clear that grpc- is a special

**Comment by zpencer** [comment]
@carl-mastrangelo PTAL

**Pull request merged**